datafusion/src/logical_plan/builder.rs

```diff
@@ -289,6 +288,37 @@ impl LogicalPlanBuilder {
         }))
     }
 
+    /// Apply a window
+    pub fn window(
+        &self,
+        window_expr: impl IntoIterator<Item = Expr>,
+        // partition_by_expr: impl IntoIterator<Item = Expr>,
+        // order_by_expr: impl IntoIterator<Item = Expr>,
+        // window_frame: Option<WindowFrame>,
```

Comment:
I think SQL also offers a "FILTER" clause for window aggregates as well, so we may want to leave room for that in the API

```suggestion
        // filter: impl IntoIterator<Item = Expr>,
        // partition_by_expr: impl IntoIterator<Item = Expr>,
        // order_by_expr: impl IntoIterator<Item = Expr>,
        // window_frame: Option<WindowFrame>,
```
This page https://sqlite.org/windowfunctions.html does a pretty good job with the syntax